### ISSUE TITLE
hostname "host-name.domainname.com" does not match the server certificate

### ISSUE THREAD

**Issue by generationtech** [SUSE/Portus#1388]
Hi, I'm trying to use the examples/compose style of implementing a Portus and Docker registry on the same host using SSL with certificates from Letsencrypt. I think I have most everything working and can get to the initial Portus setup screen and create an admin account.

But when I try to add the registry with SSL, I get the webpage error `SSL error while communicating with the registry ...` In the server console where I ran `docker-compose up` I get this error:

```
portus_1    | Started POST "/admin/registries" for 172.18.0.6 at 2017-08-16 17:56:09 +0000
portus_1    | Processing by Admin::RegistriesController#create as HTML
portus_1    |   Parameters: {"utf8"=>"✓", "authenticity_token"=>"tTdKXBrJAhSPJF1Q5hICfBJwUz1AkWBcu+K2wW0Tg1OKp/8Fu50qd2PVh9cV4B/nXorxZAAQldOcRQJJxbSnTg==", "registry"=>{"name"=>"wing_docker", "hostname"=>"host-name.domainname.com", "use_ssl"=>"1", "external_hostname"=>""}, "commit"=>"Create"}
portus_1    | hostname "host-name.domainname.com" does not match the server certificate
portus_1    | 
portus_1    | Registry not reachable:
portus_1    | #<Registry id: nil, name: "wing_docker", hostname: "host-name.domainname.com", created_at: nil, updated_at: nil, use_ssl: true, external_hostname: "">
portus_1    | SSL error while communicating with the registry, check the server logs for more details.
```
On the actual Docker registry/Portus host, the hostname `host-name.domainname.com` is hard-coded to the internal private IP address into the `hosts` file. All the testing I'm currently doing is on another internal host which also has hard-coded hostname to internal private IP address of the Portus/Registry host. Although probably not relevant, externally it's referenced correctly by outside DNS to the public IP. 

In the `.env` file, I have a line `MACHINE_FQDN=host-name.domainname.com`

And my `nginx.conf` file has this part:
```
  server {
    listen 443 ssl http2;
    server_name host-name.domainname.com;
    root /srv/Portus/public;
```
For Letsencrypt, I ran this command:
`certbot certonly --manual -d host-name.domainname.com --preferred-challenges dns`

Copied the `fullchain1.pem` to `./secrets/portus.crt`
Copied the `privkey1.pem` to `./secrets/portus.key`

My `docker-compose.yml` is default

Thanks for any help. I'm really stumped as to how to get more debugging information out of this stack

**Comment by generationtech** [comment]
It appears to do the same thing even with a self-signed certificate created like this:

`openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout portus.key -out portus.crt`

**Comment by generationtech** [comment]
This is still an issue even after attempting the cached version of head mentioned in [1386](https://github.com/SUSE/Portus/issues/1386)

**Comment by generationtech** [comment]
I've isolated the source code for this error down to:

`Portus-master/lib/portus/http_helpers.rb`

```
    def get_response_token(uri, req)
      options = { use_ssl: uri.scheme == "https", open_timeout: 2 }

      Net::HTTP.start(uri.hostname, uri.port, options) do |http|
        http.request(req)
      end
    end
```

**If I change this to:**

```
    def get_response_token(uri, req)
      options = { use_ssl: uri.scheme == "https", open_timeout: 2, verify_mode: OpenSSL::SSL::VERIFY_NONE }

      Net::HTTP.start(uri.hostname, uri.port, options) do |http|
        http.request(req)
      end
    end
```

It still doesn't work and I get this error:
```
portus_1    | Registry not reachable:
portus_1    | #<Registry id: nil, name: "wing_docker", hostname: "host-name.domainname.com", created_at: nil, updated_at: nil, use_ssl: true, external_hostname: "">
portus_1    | Error: registry does not implement v2 of the API.
```
**If I try to pull up the URL directly in a browser:**
`https://host-name.domainname.com/v2/`

I get authentication error:
```
{"errors":[{"code":"UNAUTHORIZED","message":"authentication required","detail":null}]}
```
Which I guess is kind of expected because we are trying to implement authentication with the Docker registry.

Anyone have any insight on where to go next? Either I have something wrong with the (mostly) default implementation of the Portus example production docker-compose setup or their is mistake in the current Github code.

**Comment by generationtech** [comment]
I'm not sure what's going on with the Portus project. I see lots of prior commits and contributors, and the website/documentation looks very appealing. To me the instructions are inaccurate and overall deployment strategies seem inconsistent and incongruent. Also there appears to be an exceptional amount of open Github issues.

Ultimately, I was not able to get any of the configurations in the example source dir to work with current codebase, nor could I find any helpful resources.

I did, however, discover the VMware harbor private registry system which installed correctly the first time and runs just fine for my purposes. I had hoped to use something more aligned with OSS principles, but I'll have to pass on Portus for now and take a look at it again in the future to see if things have gelled a bit better here.

**Comment by mssola** [comment]
Hi @generationtech. Sorry for taking so long to reply back (you got me on my vacations and then I was caught up with other high pressing issues...).

Regarding your issue, this is a known one (I just created #1418 to track all the duplicates). The current pipeline is broken due to quite some upgrades on gems and ruby. This is to be expected for `opensuse/portus:head`, but as you are mentioning this results in a really bad experience for newcomers. We'll try to keep the `master` branch more stable in the future...

Once #1418 is fixed I'll go back to all the duplicates and check if they are fixed.

Sorry for the inconvenience 😞 ...

**Comment by mssola** [comment]
This is fixed, see my comments on #1418.
